include/picongpu/fields/currentDeposition/Strategy.def

```diff
@@ -0,0 +1,152 @@
+     * The device local domain of fieldJ will be decomposed with a checker board.
+     *
+     * Suggestion: Use this strategy if atomic operations to global memory are slow.
+     * To utilize the device fully you should have enough supercells
+     *   - 2D: minimum multiprocessor count * 9 * 4
+     *   - 3D: minimum multiprocessor count * 27 * 4
+     */
+    struct StridedCachedSupercells
+    {
+        static constexpr bool useBlockCache = true;
+        static constexpr bool stridedMapping = true;
+        using BlockReductionOp = nvidia::functors::Atomic<
+            ::alpaka::atomic::op::Add,
+            ::alpaka::hierarchy::Threads
+        >;
+        using GridReductionOp = nvidia::functors::Add;
+    };
+
+    /** Local caching strategy
+     *
+     * The current for each particle will be reduced with atomic operations into a supercell
+     * local cache. The cache will be flushed with atomic operations to the global memory.
+     *
+     * Suggestion: Use this strategy if block local and global atomics are fast.
+     */
+    struct CachedSupercells
+    {
+        static constexpr bool useBlockCache = true;
+        static constexpr bool stridedMapping = false;
+        using BlockReductionOp = nvidia::functors::Atomic<
+            ::alpaka::atomic::op::Add,
+            ::alpaka::hierarchy::Threads
+        >;
+        using GridReductionOp = nvidia::functors::Atomic<
+            ::alpaka::atomic::op::Add,
+            ::alpaka::hierarchy::Blocks
+        >;
+    };
+
+    /** Non cached strategy
+     *
+     * The current for each particle will be reduced with atomic operations directly
+     * to the global memory.
+     *
+     * Suggestion: Use this strategy if global atomics are fast and random memory access
+     * to a wide are in the memory is not a bottle neck.
+     */
+    struct NonCachedSupercells
+    {
+        static constexpr bool useBlockCache = false;
+        static constexpr bool stridedMapping = false;
+        using BlockReductionOp = nvidia::functors::Atomic<
+            ::alpaka::atomic::op::Add,
+            ::alpaka::hierarchy::Blocks
+        >;
+        // dummy which produces a compile time error if used
+        using GridReductionOp = void;
+    };
+
+} // namespace strategy
+
+namespace traits
+{
+
+    /** Get current deposition strategy from a solver
+     *
+     * @tparam T_Solver type to derive the strategy
+     * @treturn ::type strategy description
+     */
+    template< typename T_Solver >
+    struct GetStrategy;
+
+    /** Get current deposition strategy from a solver
+     *
+     * @see GetStrategy
+     */
+    template< typename T_Solver >
+    using GetStrategy_t = typename GetStrategy< T_Solver >::type;
+
+    /** Default strategy for the current deposition
+     *
+     * Default will be selected based on the cupla accelerator.
+     *
+     * @tparam T_Acc the accelerator type
+     */
+    template<
+        typename T_Acc = cupla::AccThreadSeq
+    >
```

Comment:
Is this intended to be used with the template parameter by default? Seems like it is accelerator-specific and not clear if default makes sense.

Reply:
Yes, I could also introduce a second trait without arguments to hide the template but IMO it is not required because this trait is only used for the current deposition.
The default is always the selected cupla accelerator. If no other specialization for `GetDefaultStrategy` is given than we will always use ` strategy::StridedCachedSupercells`. Few lines below we change the default behavior for the CUDA acceleator.